src/OpenTelemetry/Trace/TracerProviderBuilder.cs

```diff
@@ -47,7 +48,7 @@ internal TracerProviderBuilder()
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
         public TracerProviderBuilder SetSampler(Sampler sampler)
         {
-            this.Sampler = sampler ?? throw new ArgumentNullException(nameof(sampler));
+            this.sampler = sampler;
```

Comment:
Double-checking, we want to allow users to set sampler to null?

Reply:
At least for now as we're seeing big memory allocation/perf difference [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1034/files).

With `sampler == null`, allocation is 656 bytes.
With `sampler != null`, allocation is 1248 bytes.

Once we solved the allocation issue in sampler, we can explore if we want to add the validation back (and remove all the `sampler?` logic from the SDK).

Reply:
yes.